src/EventStore.Core/Services/Gossip/GossipServiceBase.cs

```diff
@@ -168,10 +168,18 @@ public abstract class GossipServiceBase : IHandle<SystemMessage.SystemInit>,
 
 			message.Envelope.ReplyWith(new GossipMessage.SendGossip(_cluster, _memberInfo.HttpEndPoint));
 
-			if (_cluster.HasChangedSince(oldCluster))
+			if (_cluster.HasChangedSince(oldCluster)) {
```

Comment:
Something seems to have affected the check for whether a cluster has changed.
I see the full `CLUSTER HAS CHANGED` log every time gossip occurs when I've got a node running a different version to the others, even though nothing in the cluster has actually changed.

Reply:
Made changes so that version will be updated only when communicating with the peer directly i.e. when a node receives a gossip, it will only update version of the gossip sender node.
Since version info is not critically important and mostly static, this should be fine.

Reply:
Changed the behavior : 

1. If gossip is from new node, I will update version info from gossip
2. If gossip is from old node, I will retain previous version info